.github/workflows/ci.yml

```diff
@@ -9,66 +9,75 @@ on:
       - master
 
 jobs:
-  test:
-    runs-on: ubuntu-latest
+  ci:
+    runs-on: ubuntu-20.04
     strategy:
       fail-fast: false
       matrix:
-        output_type: [directory, tar, gpt_ext4, gpt_xfs, gpt_btrfs, gpt_squashfs, plain_squashfs]
+        distro:
+          - arch
+          - centos
+          - centos_epel
+          - clear
+          - debian
+          - ubuntu
+        format:
+          - directory
+          - tar
+          - gpt_ext4
+          - gpt_xfs
+          - gpt_btrfs
+          - gpt_squashfs
+          - plain_squashfs
+
     steps:
     - uses: actions/checkout@v2
     - uses: actions/setup-python@v1
       with:
         python-version: '3.x'
-    - name: Update package cache
-      run: sudo apt-get --assume-yes update
+
     - name: Install dependencies
-      run: sudo apt-get --assume-yes install systemd-container squashfs-tools e2fsprogs xfsprogs libzstd1-dev libgpgme-dev libcurl4-openssl-dev asciidoc
-    - name: Download libarchive source
-      run: |
-        wget "https://github.com/libarchive/libarchive/releases/download/v3.4.3/libarchive-3.4.3.tar.gz"
```

Comment:
humn.. so now in ubuntu 20.04 we have zstd?

Reply:
Yup, It was only released to github actions a month ago so I'm kinda lucky that I look at it now since all this stuff can be simplified.

The bad thing is yum doesn't exist anymore in focal repos so the centos builds are broken. I'm seeing if I can bootstrap dnf on ubuntu but it's a pain. Ridiculously complex compared to pacman imo.